### ISSUE TITLE
pdns-recursor bug: will use IPv4 to resolve even if configured to not use

### ISSUE THREAD

**Issue by stasic** [PowerDNS/pdns#6099]
### Short description
I wanted to run pdns-recursor only in IPv6 mode, but it also uses IPv4

### Environment

 - Operating system: ubuntu 17.10
 - Software version: 4.0.6-1
 - Software source: Operating system repository

### Steps to reproduce
1. 
```
grep -v '^#' /etc/powerdns/recursor.conf
allow-from=2001:62a:4:203::/48
config-dir=/etc/powerdns
dnssec=validate
hint-file=/usr/share/dns/root.hints
local-address=2001:62a:4:203::bca5
local-port=53
query-local-address=
query-local-address6=2001:62a:4:203::bca5
quiet=yes
security-poll-suffix=
setgid=pdns
setuid=pdns
```
2. 
```
strace -p 25556 -f -e trace=network
strace: Process 25556 attached with 4 threads
[pid 25557] recvmsg(4, {msg_name={sa_family=AF_INET6, sin6_port=htons(38166), inet_pton(AF_INET6, "2001:62a:4:203::bca5", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, msg_namelen=28, msg_iov=[{iov_base="\247\241\1 \0\1\0\0\0\0\0\1\3www\6google\3com\0\0\34\0\1"..., iov_len=1500}], msg_iovlen=1, msg_control=[{cmsg_len=20, cmsg_level=SOL_SOCKET, cmsg_type=0x1d /* SCM_??? */}], msg_controllen=20, msg_flags=0}, 0) = 55
[pid 25557] recvmsg(4, {msg_namelen=28}, 0) = -1 EAGAIN (Resource temporarily unavailable)
[pid 25559] socket(AF_INET6, SOCK_DGRAM, IPPROTO_IP) = 22
[pid 25559] bind(22, {sa_family=AF_INET6, sin6_port=htons(28282), inet_pton(AF_INET6, "2001:62a:4:203::bca5", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
[pid 25559] connect(22, {sa_family=AF_INET6, sin6_port=htons(53), inet_pton(AF_INET6, "2001:7fd::1", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
[pid 25559] send(22, "\3p\0\0\0\1\0\0\0\0\0\1\3www\6google\3com\0\0\34\0\1"..., 43, 0) = 43
[pid 25559] recvfrom(22, "\3p\200\0\0\1\0\0\0\17\0\33\3www\6google\3com\0\0\34\0\1"..., 1680, 0, {sa_family=AF_INET6, sin6_port=htons(53), inet_pton(AF_INET6, "2001:7fd::1", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, [28]) = 1174
[pid 25559] socket(AF_INET6, SOCK_DGRAM, IPPROTO_IP) = 23
[pid 25559] bind(23, {sa_family=AF_INET6, sin6_port=htons(47581), inet_pton(AF_INET6, "2001:62a:4:203::bca5", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
[pid 25559] connect(23, {sa_family=AF_INET6, sin6_port=htons(53), inet_pton(AF_INET6, "2001:503:a83e::2:30", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
[pid 25559] send(23, "y\322\0\0\0\1\0\0\0\0\0\1\3www\6google\3com\0\0\34\0\1"..., 43, 0) = 43
[pid 25559] recvfrom(23, "y\322\200\0\0\1\0\0\0\10\0\5\3www\6google\3com\0\0\34\0\1"..., 1680, 0, {sa_family=AF_INET6, sin6_port=htons(53), inet_pton(AF_INET6, "2001:503:a83e::2:30", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, [28]) = 664
[pid 25559] socket(AF_INET, SOCK_DGRAM, IPPROTO_IP) = 22
[pid 25559] bind(22, {sa_family=AF_INET, sin_port=htons(9768), sin_addr=inet_addr("0.0.0.0")}, 16) = 0
[pid 25559] connect(22, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("216.239.38.10")}, 16) = 0
[pid 25559] send(22, "\374\"\0\0\0\1\0\0\0\0\0\1\3www\6google\3com\0\0\34\0\1"..., 43, 0) = 43
[pid 25559] recvfrom(22, "\374\"\204\0\0\1\0\1\0\0\0\0\3www\6google\3com\0\0\34\0\1"..., 1680, 0, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("216.239.38.10")}, [28->16]) = 60
[pid 25559] socket(AF_INET, SOCK_DGRAM, IPPROTO_IP) = 23
[pid 25559] bind(23, {sa_family=AF_INET, sin_port=htons(62231), sin_addr=inet_addr("0.0.0.0")}, 16) = 0
[pid 25559] connect(23, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("216.239.36.10")}, 16) = 0
[pid 25559] send(23, "V3\0\0\0\1\0\0\0\0\0\1\3www\6google\3com\0\0\2\0\1"..., 43, 0) = 43
[pid 25559] recvfrom(23, "V3\204\0\0\1\0\0\0\1\0\0\3www\6google\3com\0\0\2\0\1"..., 1680, 0, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("216.239.36.10")}, [28->16]) = 82
[pid 25559] socket(AF_INET6, SOCK_DGRAM, IPPROTO_IP) = 22
[pid 25559] bind(22, {sa_family=AF_INET6, sin6_port=htons(49363), inet_pton(AF_INET6, "2001:62a:4:203::bca5", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
[pid 25559] connect(22, {sa_family=AF_INET6, sin6_port=htons(53), inet_pton(AF_INET6, "2001:500:856e::30", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, 28) = 0
[pid 25559] send(22, "\37;\0\0\0\1\0\0\0\0\0\1\3com\0\0000\0\1\0\0)\6\220\0\0\200\0\0\0", 32, 0) = 32
[pid 25559] recvfrom(22, "\37;\204\0\0\1\0\3\0\0\0\1\3com\0\0000\0\1\300\f\0000\0\1\0\1Q\200\0"..., 1680, 0, {sa_family=AF_INET6, sin6_port=htons(53), inet_pton(AF_INET6, "2001:500:856e::30", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, [28]) = 743
[pid 25559] socket(AF_INET, SOCK_DGRAM, IPPROTO_IP) = 23
[pid 25559] bind(23, {sa_family=AF_INET, sin_port=htons(10366), sin_addr=inet_addr("0.0.0.0")}, 16) = 0
[pid 25559] connect(23, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("192.26.92.30")}, 16) = 0
[pid 25559] send(23, "\276\32\0\0\0\1\0\0\0\0\0\1\6google\3com\0\0+\0\1\0\0)\6"..., 39, 0) = 39
[pid 25559] recvfrom(23, "\276\32\204\0\0\1\0\0\0\6\0\1\6google\3com\0\0+\0\1 CK0"..., 1680, 0, {sa_family=AF_INET, sin_port=htons(53), sin_addr=inet_addr("192.26.92.30")}, [28->16]) = 760
[pid 25559] sendmsg(4, {msg_name={sa_family=AF_INET6, sin6_port=htons(38166), inet_pton(AF_INET6, "2001:62a:4:203::bca5", &sin6_addr), sin6_flowinfo=htonl(0), sin6_scope_id=0}, msg_namelen=28, msg_iov=[{iov_base="\247\241\201\200\0\1\0\1\0\0\0\1\3www\6google\3com\0\0\34\0\1"..., iov_len=71}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, 0) = 71
^Cstrace: Process 25556 detached
strace: Process 25557 detached
strace: Process 25558 detached
strace: Process 25559 detached
```

### Expected behaviour
just resolving over IPv6

### Actual behaviour
it resolves over IPv4 and IPv6

### Other information
workaround: use config option:
```
dont-query:0.0.0.0/0
```

**Comment by Habbie** [comment]
```
  if(family==AF_INET) {
    if(g_localQueryAddresses4.empty())
      ret = g_local4;
    else
      ret = g_localQueryAddresses4[dns_random(g_localQueryAddresses4.size())];
    ret.sin4.sin_port = htons(port);
  }
```

For v6, this is guarded by `SyncRes::s_doIPv6`, but we have no such bool for v4.

**Comment by omoerbeek** [comment]
Should be fixed by #9196
